scripts/download-vscode.js

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env node
+
+const ncp = require('ncp');
+const path = require('path');
+const fs = require('fs');
+const downloadAndUnzipVSCode = require('vscode-test').downloadAndUnzipVSCode;
+
+// Copy vscode to these directories
+const extensionDirectories = [
+  'salesforcedx-vscode-apex',
+  'salesforcedx-vscode-apex-debugger',
+  'salesforcedx-vscode-apex-replay-debugger',
+  'salesforcedx-vscode-core',
+  'salesforcedx-vscode-lightning',
+  'salesforcedx-vscode-lwc',
+  'salesforcedx-vscode-visualforce'
+];
+
+// Executable path looks something like:
+// ~/salesforcedx-vscode/packages/salesforcedx-vscode-lwc/.vscode-test/vscode-1.41.1/Visual Studio Code.app/Contents/MacOS/Electron
+downloadAndUnzipVSCode()
+  .then(executablePath => {
+    console.log('Executable Path: ' + executablePath);
+
+    let vscodeIndex = executablePath.indexOf('.vscode-test') + 13;
+
+    // 'vscode-1.41.1'
+    let vscodeDirname = executablePath.substring(
+      vscodeIndex,
+      executablePath.indexOf(path.sep, vscodeIndex)
+    );
+    process.env.CODE_VERSION = vscodeDirname;
+
+    // '~/salesforcedx-vscode/.vscode-test/vscode-1.41.1/'
+    let vscodeFullPath = executablePath.substring(
+      0,
+      vscodeIndex + vscodeDirname.length + 1
+    );
+
+    // '~/salesforcedx-vscode/'
+    let vscodeBasePath = path.dirname(path.dirname(vscodeFullPath));
+
+    console.log('Directory Name: ' + vscodeDirname);
+    console.log('Base Path: ' + vscodeBasePath);
+    console.log('Full Path: ' + vscodeFullPath);
+
+    // TODO make this configurable as per package or global?
+
+    // For each extension, copy over the vscode binary
+    for (let i = 0; i < extensionDirectories.length; i++) {
```

Comment:
I know that this resembles how running integration test work right now but did we look at just downloading vscode at a top level location and then just use that vscode to run all the tests ?

Reply:
I didn't try having them all point to the top level directory, but this is better than what currently happens where we download the binary and extract it in every package. Now it just downloads it once and extracts it everywhere

Reply:
Ok, let's log a work item to try that approach later. No need to change it as part of this PR.